### PR TITLE
feat: Port LayoutRound code from WinUI

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): So far, Skia only part of #12082

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 409ac22</samp>

This pull request implements and refactors the layout rounding feature for Uno, which improves the visual quality of UI elements by aligning them to pixel boundaries. It adds and modifies code in the `LayoutHelper`, `UIElement`, `Border`, `CalendarViewBaseItem`, and `TextBlock` classes, and updates the corresponding files. It also improves the code formatting and removes some unused or redundant methods.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/LayoutHelper.cs [hunk 1]
```diff
@@ -52,6 +52,21 @@ internal static (Size min, Size max) GetMinMax(this IFrameworkElement e)
 				.AtMost(maxSize)
 				.AtLeast(minSize); // UWP is applying "min" after "max", so if "min" > "max", "min" wins
 
+			if (e is UIElement uiElement && uiElement.GetUseLayoutRounding())
+			{
+				// It is possible for max vars to be INF so be don't want to round those.
+
+				minSize.Width = uiElement.LayoutRound(minSize.Width);
+
+				if (double.IsFinite(maxSize.Width))
```
Comment (jeromelaban) [inline src/Uno.UI/UI/LayoutHelper.cs:61]: The 🤖 is right :)

File: src/Uno.UI/UI/Xaml/Controls/Border/Border.Helpers.cs [hunk 2]
```diff
@@ -33,29 +37,6 @@ private static Size HelperCollapseThickness(Thickness thickness)
...
-			// 2) There is no CornerRadius - we cannot guarantee snapping when there is a CornerRadius.
-			// 3) Plateau != 1.0 - this snapping is to ensure that integral values of BorderThickness don't cause subpixel rendering at high plateau.
-			// TODO: Remove check for plateau, BorderThickness should be consistently rounded at all plateaus.
-			//return (RootScale.GetRasterizationScaleForElement(element) != 1.0f) &&
-			//       element.GetUseLayoutRounding() &&
-			//       (!HasNonZeroCornerRadius(element.GetCornerRadius()));
-			return (RootScale.GetRasterizationScaleForElement(element) != 1.0f) &&
-				   element.GetUseLayoutRounding() &&
-				   (!HasNonZeroCornerRadius(element.GetCornerRadius()));
-		}
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Border/Border.Helpers.cs:49]: This was a particularly sensitive behavior that was added, especially on iOS. We'll need to validate that this still works properly. Make sure to validate with related issues when blaming this function.

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/Border/Border.Helpers.cs:49]: @jeromelaban iOS shouldn't be affected. `GetUseLayoutRounding()` is always false for iOS.

https://github.com/unoplatform/uno/blob/58b7d854aacf85bc71a701040651a99c419d5615/src/Uno.UI/UI/Xaml/UIElement.cs#L1105-L1112

File: src/Uno.UI/UI/LayoutHelper.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI/UI/LayoutHelper.cs:61]: Not sure what you mean

Comment (MartinZikmund) [reply src/Uno.UI/UI/LayoutHelper.cs:61]: Probably for older version of the code.

File: src/Uno.UI/UI/Xaml/Controls/Border/Border.Helpers.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Border/Border.Helpers.cs:49]: Even previously the `GetUseLayoutRounding` was always `false` so this shouldn't change anything on non-Skia targets

**Comment by jeromelaban** [comment]
@Youssef1313 @MartinZikmund did you review the screenshot tests on all platforms?

**Comment by Youssef1313** [comment]
Quickly went through them yes. The only meaningful change I could find is a bug fix.

**Pull request merged**